MicroPython/src/communication/wirerless_connection_controller.py

```diff
@@ -12,7 +11,7 @@ class WirelessConnectionController:
     Wifi handler class.
     """
     WIFI_CONNECTION_CHECK_SLEEP_TIME_S = 5
-    WIFI_CONNECTION_MAX_NUMBER_OF_RETRIES = 10
+    WIFI_CONNECTION_MAX_NUMBER_OF_RETRIES = 3
```

Comment:
Not sure if 3 attempts are even needed? Maybe just try once and if it doesn't work move to the next one.